### DISCUSSION TITLE
Improve server's async runtime/executor from actix_rt/tokio to io_uring

### DISCUSSION THREAD

**Discussion by pymongo** [tensorbase/tensorbase#144] [Ideas]
https://github.com/tensorbase/tensorbase/blob/ce38fb3fd63e0c86fb99ff1aaacc50367d33dbbf/crates/server/src/server.rs#L15-L19

I am using glommio(io_uring) on my websocket chat server, it works great, consider using glommio on tensorbase's server?

https://github.com/DataDog/glommio

**Comment by jinmingjian** [comment]
@pymongo cool! does glommio support fallback?  although io_uring is great, but it needs new kernel support. For example, some use the Ubuntu 18.04. It is possible to firstly investigate the benefits of moving to io_uring.

**Comment by jinmingjian** [comment]
[CVE-2021-20226](https://bugzilla.redhat.com/show_bug.cgi?id=1873476#c16) kernel: use-after-free in io_uring feature

**Comment by pymongo** [comment]
tokio-uring v0.1 is release, tokio-uring io_uring executor is compatible with hyper web application in my computer

**Comment by pymongo** [comment]
I think runtime/executor should decoupling, user can choose different runtime, eg. on Linux can use io_uring(tokio-uring/glommio), on mac can use tokio

**Reply by jinmingjian** [reply]
thanks for great infos! I've heared the progress of io_uring in tokio. It is possible. Although cve out, we can let use to choose. How about your decoupling? It is great to have generally I guess:)

**Reply by pymongo** [reply]
Here is my PR to a hyper applications to support io_uring https://github.com/KernelErr/blacktea/pull/1/files

by hyper API: `hyper::server::conn::Http::new().with_executor`
